scripts/buf_download.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+# Download buf
+download_buf() {
+  BUFVERSION="v0.37.1"
+  echo Using Bufbuild version $BUFVERSION
+  curl -sSLOC - https://github.com/bufbuild/buf/releases/download/$BUFVERSION/buf-$(uname -s)-$(uname -m)
+  chmod a+x buf-$(uname -s)-$(uname -m)
+  echo Buf file downloaded
+}
+
+if [[ "$OSTYPE" == "linux-gnu"* ]]; then
+  download_buf
+elif [[ "$OSTYPE" == "darwin"* ]]; then
+  download_buf
+elif [[ "$OSTYPE" == "freebsd"* ]]; then
```

Comment:
Is it good to have OS check for these 3 types:
1. linux-gnu
2. darwin
3. freebsd

Or are we missing anything here or should we remove `freebsd` check as it might comes under Linux? 

@FareesHussain any thoughts on this?

Reply:
I've seen some scripts where `freebsd` is used besides `linux-gnu` so I think we need to use `freebsd`.

And I think we've covered everything required.

Reply:
Thanks for confirming.